### PR TITLE
Add ChecksSelection component

### PR DESCRIPTION
**Pull request by dottorblaster**
Just externalized the checks selection into a component of its own. Stories available :metal:

### CODE REVIEW

**Comment by dottorblaster** [comment]
@nelsonkopliku @EMaksy I will be away for a couple weeks, so I'll leave some notes on this to make the merge and the iteration (if any) easiest as it can be.

- The logic that displays the selected checks has been simplified a bit
- You can use the onClear handler to clear any redux state that holds any saga result. It gets called every time a check gets selected, and on mount
- The component is completely drop-in: remaining stuff is using this to build the actual hosts check result page and dropping this into the old clusters checks selection page, making the same component declared locally into that directory (`ClusterDetails`) useless (and the directory itself slimmer).

Right now I don't have any other remark in my mind but feel free to ask!

File: assets/js/components/ChecksSelection/ChecksSelection.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    onSave: { action: 'Save' },
+    onClear: {
+      action: 'Clear',
+      description:
+        'Gets called on mount and when checks are selected. It can be used to clear any external state.',
+    },
+  },
+};
+
+export const Default = {
```
Comment (arbulu89) [inline assets/js/components/ChecksSelection/ChecksSelection.stories.jsx:86]: We could maybe have an entry with already selected checks

File: assets/js/components/ChecksSelection/ChecksSelection.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  args: {
+    catalog,
+    resourceID,
+  },
+};
+
+export const Loading = {
+  args: {
+    catalog,
```
Comment (arbulu89) [inline assets/js/components/ChecksSelection/ChecksSelection.stories.jsx:95]: You could have to use the default values:
```
args: {
   ...Default,
   loading: true,
}
```

**Review by arbulu89 — APPROVED** [review]
Fantastic job man!
Eager to see how it will be applied in the cluster and host views :slightly_smiling_face:

**Comment by nelsonkopliku** [comment]
@arbulu89 addressed the feedbacks and took the opportunity to align naming from `resource` to `target` in 27e715ae5a9fcb39b5c08d204a5ba56d97cd91af

File: assets/js/components/ChecksSelection/ChecksSelection.stories.jsx (hunk 1 above)
Comment (arbulu89) [reply assets/js/components/ChecksSelection/ChecksSelection.stories.jsx:95]: In fact, i liked the default empty. I would've added a new story :see_no_evil:

**Review by arbulu89 — APPROVED** [review]
Thank you @nelsonkopliku 
:heart: 
PD: Still added a comment XD, but no need of a new review, the code looks great

**Pull request merged**